### DISCUSSION TITLE
Tremorsense

### DISCUSSION THREAD

**Discussion by Kamarathin** [dmdorman/hero6e-foundryvtt#504] [General]
Hi folks,

So Foundry has Tremorsense as a Vision Mode, and Feel Tremor as a Detection mode. The former is blocked by walls and be set to be directional (i.e. the normal 120 degree vision for Hero System) but is blocked by walls. The latter is 360 degrees and not blocked by walls.

I have a PC that has Tremorsense built with Spatial Awareness (Touch Group) but only works through stone/earth etc. I'd like to be able to put up walls that block the characters tremorsense to indicate that they are aren't made of stone and/or designate locations within the scene that are not visible with tremorsense, like wooden floors.

Do you know of any way to do that?

**Comment by aeauseth** [comment]
We don't have much in the way of ENHANCED SENSES supported yet.  @phBalance is working on some flash stuff with improved sense support, but vision/sense/wall interactions is complicated and we don't have the framework in place to support it yet.  

We have discussed how we might implement it in the future.  Material property types for walls.  How each material type handles sense group/modifiers.  

I don't really have much of a temporary solution for you, other than to try and use the built in FoundryVTT visions and/or modifying wall vision during the game to accommodate for unique sense properties. 

We don't have a good concept for ARC OF PERCEPTION yet either.  We tried limiting token vision to 120 degrees, but it was awkward during gameplay as you have to rotate the token to see your target or even a wall/car/object.   While perhaps realistic, it lacked a sense of immersion.
